src/main/java/org/openlmis/stockmanagement/controller/StockCardController.java

```diff
@@ -205,12 +211,30 @@ public ResponseEntity adjustStock(@PathVariable long facilityId,
 
             // create entry from event
             long quantity = event.getQuantity();
-            quantity = reason.getAdditive() ? quantity : quantity * -1;
-            StockCardEntry entry = new StockCardEntry(card,
-                StockCardEntryType.ADJUSTMENT,
-                quantity);
+            if (null != reason) {
```

Comment:
StockEvent.getQuantity() already has a bit of logic related to the sign (positive or negative) of the value it returns. The rest of this logic is generic and useful enough that it probably belongs in StockEvent as well.